devicetypes/widomsrl/widom-smart-dry-contact.src/widom-smart-dry-contact.groovy

```diff
@@ -0,0 +1,340 @@
+	def version = cmdVersions()[cmd.commandClass as Integer]
+	def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+	def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+	if (encapsulatedCommand) {
+		logging("Parsed Crc16Encap into: ${encapsulatedCommand}")
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		logging("Unable to extract CRC16 command from $cmd","warn")
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand(cmdVersions())
+	if (encapsulatedCommand) {
+		logging("Parsed MultiChannelCmdEncap ${encapsulatedCommand}")
+		zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+	} else {
+		logging("Unable to extract MultiChannel command from $cmd","warn")
+	}
+}
+
+private logging(text, type = "debug") {
+	if (settings.logging == "true" || type == "warn") {
+		log."$type" "${device.displayName} - $text"
+	}
+}
+
+private secEncap(physicalgraph.zwave.Command cmd) {
+	logging("encapsulating command using Secure Encapsulation, command: $cmd","info")
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private crcEncap(physicalgraph.zwave.Command cmd) {
+	logging("encapsulating command using CRC16 Encapsulation, command: $cmd","info")
+	zwave.crc16EncapV1.crc16Encap().encapsulate(cmd).format()
+}
+
+private multiEncap(physicalgraph.zwave.Command cmd, Integer ep) {
+	logging("encapsulating command using MultiChannel Encapsulation, ep: $ep command: $cmd","info")
+	zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:ep).encapsulate(cmd)
+}
+
+private encap(physicalgraph.zwave.Command cmd, Integer ep) {
+	encap(multiEncap(cmd, ep))
+}
+
+private encap(physicalgraph.zwave.Command cmd) {
+	if (zwaveInfo.zw.contains("s")) {
+		secEncap(cmd)
+	} else if (zwaveInfo.cc.contains("56")){
+		crcEncap(cmd)
+	} else {
+		logging("no encapsulation supported for command: $cmd","info")
+		cmd.format()
+	}
+}
+
+private encapSequence(cmds, Integer delay=250) {
+	delayBetween(cmds.collect{ encap(it) }, delay)
+}
+
+private encapSequence(cmds, Integer delay, Integer ep) {
+	delayBetween(cmds.collect{ encap(it, ep) }, delay)
+}
+
+private List intToParam(Long value, Integer size = 1) {
+	def result = []
+	size.times {
+		result = result.plus(0, (value & 0xFF) as Short)
+		value = (value >> 8)
+	}
+	return result
+}
+
+/*
+##########################
+## Device Configuration ##
+##########################
+*/
+private Map cmdVersions() {
+	[0x5E: 2, 0x86: 2, 0x72: 2, 0x59: 2, 0x98: 1, 0x25: 1, 0x5A: 1, 0x85: 2, 0x70: 1, 0x8E: 2, 0x6C: 1] 
+}
+
+private parameterMap() {[
+		[key: "numClickLoad", num: 1, size: 1, type: "number", min: 0, max: 7, def: 7, title: "Numbers of clicks to controlthe loads",
+			descr: "Define which sequences of clicks control the load (see device manual)."],     
```

Comment:
Please remove those unnecessary 4 spaces at the end of the line.

Reply:
fixed